packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -71,11 +71,8 @@ const schema: SchemaType<DbComment> = {
     type: String,
     optional: true,
     canRead: [documentIsNotDeleted],
-    onInsert: async (document, currentUser) => {
-      // if userId is changing, change the author name too
-      if (document.userId) {
-        return await userGetDisplayNameById(document.userId)
-      }
+    onInsert: (document, currentUser) => {
+      return userGetDisplayName(currentUser);
```

Comment:
Potentially exposes an issue for unit tests, if test comments are created using a session that doesn't match the user that's supposed to be posting the comments.